pkg/guid2english/transformer.go

```diff
@@ -80,21 +84,51 @@ func New(m Mapper) *Transformer {
 	}
 }
 
+func (t *Transformer) bufferMap(match []byte) []byte {
+	// The regex only matches valid GUIDs, so this must parse.
+	g, err := guid.Parse(string(match))
+	if err != nil {
+		return match
+	}
+	return t.mapper.Map(*g)
+}
+
 // Transform implements transform.Transformer.Transform().
 func (t *Transformer) Transform(dst, src []byte, atEOF bool) (nDst, nSrc int, err error) {
-	transformed := guidRegex.ReplaceAllFunc(src, func(match []byte) []byte {
-		// The regex only matches valid GUIDs, so this must parse.
-		g, err := guid.Parse(string(match))
-		if err != nil {
-			return src
+	if atEOF {
+		// we have the end of file, try to process all at once
+		transformed := guidRegex.ReplaceAllFunc(src, t.bufferMap)
+		if len(transformed) > len(dst) {
+			// we were too optimistic, dst is too short
+			d, s, e := t.Transform(dst, src, false)
+			if e != transform.ErrShortSrc {
+				return d, s, e
+			}
+			return d, s, transform.ErrShortDst
 		}
-		return t.mapper.Map(*g)
-	})
-	if len(transformed) > len(dst) {
-		return 0, 0, transform.ErrShortDst
+		copy(dst, transformed)
+		return len(transformed), len(src), nil
+	}
+	loc := guidRegex.FindIndex(src)
```

Comment:
Doesn't this cause you to miss the partial guid if you have another GUID who's index can be found?

Reply:
I don't get your point...
FindIndex will find the first match in src from the beginning of the buffer. a partial GUID can only be at the end, and we don't consume the end of the buffer so next call to Transform by the transform.Reader will give us the remains of src appended with new data (unless at the end of file where atEOF will eventually be set which cover the case of the file containing a partial GUID at its end)

Reply:
This looks correct to me. The return is `return loc[0] + len(mappedGUID), loc[1], transform.ErrShortSrc`, so only the first guid is consumed. A partial guid wold be left in the source buffer.

Reply:
I think I misunderstood how the transform function is supposed to work, I think this is right